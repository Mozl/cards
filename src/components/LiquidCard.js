import React, { Component } from 'react';
import styled from 'styled-components';

const Box = styled.div`
  border: 0.1rem solid black;
  width: 15rem;
  display: flex;
  flex-direction: column;
  text-color: black;
  border-radius: 1.25rem;
  padding-bottom: 1rem;
  margin: 2rem;
`;

const Wrapper = styled.div``;

class LiquidCard extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    return (
      <Wrapper>
        {this.props.user.income >= 17000 && (
          <Box>
            <h2>Liquid Life</h2>
            <p>Apr: 33.9%</p>
            <p>Balance Transfer Offer Duration: 12 months</p>
            <p>Purchase Offer Duration: 6 Months</p>
            <p>Credit Available: Â£3000</p>
          </Box>
        )}
      </Wrapper>
    );
  }
}

export default LiquidCard;
