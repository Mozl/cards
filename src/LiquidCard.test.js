import React from 'react';
import ReactDOM from 'react-dom';
import LiquidCard from './components/LiquidCard';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LiquidCard />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<LiquidCard />);
  expect(tree).toMatchSnapshot();
});

it('prop selected changes', () => {
  const wrapper = mount(<LiquidCard selected={false} />);
  wrapper.setProps({ selected: true });
  expect(wrapper.props().selected).toEqual(true);
});
