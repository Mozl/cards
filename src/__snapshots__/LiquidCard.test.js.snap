// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-htpNat KtFsv"
>
  <div
    className="sc-bwzfXH bfmIOp"
  >
    <h2>
      Liquid Life
    </h2>
  </div>
</div>
`;
